src/renderers/WebGLRenderer.js

```diff
@@ -1982,10 +1982,18 @@ function WebGLRenderer( parameters ) {
 
 			}
 
-			var offset = uvScaleMap.offset;
-			var repeat = uvScaleMap.repeat;
+			if ( uvScaleMap.matrixAutoUpdate === true ) {
 
-			uniforms.offsetRepeat.value.set( offset.x, offset.y, repeat.x, repeat.y );
+				var offset = uvScaleMap.offset;
+				var repeat = uvScaleMap.repeat;
+
+				uniforms.uvTransform.value.setUvTransform( offset.x, offset.y, repeat.x, repeat.y, 0, 0, 0 );
```

Comment:
OK... you realize, however, that there may be multiple textures, and only one will have its `matrix` "updated" by the renderer. The way I wrote it, `texture.matrix` is never modified by the renderer

Are you OK with that?